bpfd/src/command.rs

```diff
@@ -231,176 +165,51 @@ impl TryFrom<AyaProgInfo> for KernelProgramInfo {
-    pub(crate) fn_name: String,
-    pub(crate) offset: u64,
-    pub(crate) retprobe: bool,
-    pub(crate) namespace: Option<String>,
-}
-
-#[derive(Debug, Clone, Deserialize, Serialize)]
-pub(crate) struct UprobeAttachInfo {
-    pub(crate) fn_name: Option<String>,
-    pub(crate) offset: u64,
-    pub(crate) target: String,
-    pub(crate) retprobe: bool,
-    pub(crate) pid: Option<i32>,
-    pub(crate) namespace: Option<String>,
-}
-
-#[derive(Debug, Eq, Ord, PartialEq, PartialOrd, Serialize, Deserialize, Clone)]
-pub(crate) struct Metadata {
-    pub(crate) priority: i32,
-    pub(crate) attached: bool,
-}
-
-impl Metadata {
-    pub(crate) fn new(priority: i32) -> Self {
-        Metadata {
-            priority,
-            attached: false,
-        }
-    }
-}
-
-#[derive(Serialize, Deserialize, Clone)]
-pub(crate) enum Program {
-    Xdp(XdpProgram),
-    Tracepoint(TracepointProgram),
-    Tc(TcProgram),
-    Kprobe(KprobeProgram),
-    Uprobe(UprobeProgram),
-}
-
-#[derive(Serialize, Deserialize, Clone)]
-pub(crate) struct XdpProgram {
-    pub(crate) data: ProgramData,
-    pub(crate) info: XdpProgramInfo,
-}
-
-impl XdpProgram {
-    pub(crate) fn new(data: ProgramData, info: XdpProgramInfo) -> Self {
-        Self { data, info }
-    }
-}
-
-#[derive(Serialize, Deserialize, Clone)]
-pub(crate) struct TcProgram {
-    pub(crate) data: ProgramData,
-    pub(crate) info: TcProgramInfo,
-}
-
-#[derive(Serialize, Deserialize, Clone)]
-pub(crate) struct TracepointProgram {
-    pub(crate) data: ProgramData,
-    pub(crate) info: TracepointProgramInfo,
-}
-
-impl TracepointProgram {
-    pub(crate) fn new(data: ProgramData, info: TracepointProgramInfo) -> Self {
-        Self { data, info }
-    }
-}
-
-#[derive(Serialize, Deserialize, Clone)]
-pub(crate) struct KprobeProgram {
-    pub(crate) data: ProgramData,
-    pub(crate) info: KprobeProgramInfo,
-}
-
-#[derive(Serialize, Deserialize, Clone)]
-pub(crate) struct UprobeProgram {
-    pub(crate) data: ProgramData,
-    pub(crate) info: UprobeProgramInfo,
-}
-
-impl KprobeProgram {
-    pub(crate) fn _new(data: ProgramData, info: KprobeProgramInfo) -> Self {
-        Self { data, info }
-    }
-}
-
-impl UprobeProgram {
-    pub(crate) fn _new(data: ProgramData, info: UprobeProgramInfo) -> Self {
-        Self { data, info }
-    }
-}
-
-// ProgramData represents all of the core information needed to load
-// a program reguardless of ProgramType.
-#[derive(Serialize, Deserialize, Clone)]
+/// ProgramInfo stores information about bpf programs that are loaded and managed
+/// by bpfd.
+#[derive(Debug, Serialize, Deserialize, Clone)]
 pub(crate) struct ProgramData {
+    // known at load time, set by user
+    pub(crate) name: String,
     pub(crate) location: Location,
-    pub(crate) section_name: String,
-    pub(crate) global_data: HashMap<String, Vec<u8>>,
-    pub(crate) owner: String,
-    pub(crate) map_owner_uuid: Option<Uuid>,
+    pub(crate) uuid: Option<Uuid>,
```

Comment:
Per comment on proto this should be a `HashMap<String, String>: metadata`

Reply:
Done